README.md

```diff
@@ -917,6 +917,62 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
 
+* <a id='comments'></a>(<a href='#comments'>link</a>) **Comment blocks should use single-line comments (`//` and `///`)**, rather than multi-line comments (`/* ... */` and `/** ... */`). [![SwiftFormat: blockComments](https://img.shields.io/badge/SwiftFormat-blockComments-7B0051.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#blockComments)
+
+  <details>
+
+  ```swift
+  // WRONG
+
+  /**
+  * A planet that exists somnewhere in the universe.
+  *
+  * Planets have many properties. For example, the best planets
+  * have atmospheres and bodies of water to support life.
+  */
+  class Planet {
+    /**
+      Terraforms the planet, by adding an atmosphere and ocean
+      that is hospitable for life.
+      - Precondition: the planet does not already have an atmosphere.
+        Adding another atmosphere to a planet that already has one is a very bad idea.
+    */
+    func terraform() {
+      /* 
+      Generate the atmosphere first, before generating the ocean.
+      Otherwise, the water will just boil off immediately.
+      */
+      generateAtmosphere()
+
+      /* Now that we have an atmosphere, it's safe to generate the ocean */
```

Comment:
Should our documentation comments wrap at our 100 column limit? I don't think we have any rule that encourages the current length that's being used

Reply:
> Should our documentation comments wrap at our 100 column limit?

One thing to consider here is that I don't think there are any auto-formatting rule implementations that handle wrapping comments

>  I don't think we have any rule that encourages the current length that's being used

What do you mean?

Reply:
> One thing to consider here is that I don't think there are any auto-formatting rule implementations that handle wrapping comments
> 
> > I don't think we have any rule that encourages the current length that's being used
> 
> What do you mean?

Reply:
Aha, I see -- this is another good question about comment style. I personally tend to manually balance comment blocks so all of the lines in a block are approximately the same length, rather than wrapping specifically at the column limit. I think I also tend to wrap comments are more like 70-80 columns by default, rather than 100 columns, which is also just a stylistic thing. I know different folks have different preferences / practices here.

I think it would be great to have consistent guidelines for comment wrapping, with auto-formatting.

Reply:
Not that all of the wrapping in the example stands up to much scrutiny lol! Some of these comments would be more pleasing if wrapped a bit differently

Reply:
Wrapping exactly at the 100 column limit looks like:

```swift
/// A planet that exists somnewhere in the universe.
///
/// Planets have many properties. For example, the best planets have atmospheres and bodies of 
/// water to support life.
class Planet {
  /// Terraforms the planet, by adding an atmosphere and ocean that is hospitable for life.
  ///  - Precondition: the planet does not already have an atmosphere. Adding another atmosphere 
  ///    to a planet that already has one is a very bad idea.
  func terraform() {
    // Generate the atmosphere first, before generating the ocean. Otherwise, the water will just 
    // boil off immediately.
    generateAtmosphere()

    // Now that we have an atmosphere, it's safe to generate the ocean
    generateOceans()
  }
}
```

a slightly improved version of the existing wrapping could like like this:

```swift
/// A planet that exists somnewhere in the universe.
///
/// Planets have many properties. For example, the best planets
/// have atmospheres and bodies of water to support life.
class Planet {
  /// Terraforms the planet, by adding an atmosphere and ocean that is hospitable for life.
  ///  - Precondition: the planet does not already have an atmosphere.
  ///    Adding another atmosphere to a planet that already has one is a very bad idea.
  func terraform() {
    // Generate the atmosphere first, before generating the ocean.
    // Otherwise, the water will just boil off immediately.
    generateAtmosphere()

    // Now that we have an atmosphere, it's safe to generate the ocean
    generateOceans()
  }
}
```

Reply:
I think it would be really useful in the future to have a rule with autocorrection around consistent comment wrapping. I try to have my comments be <= 100 lines but it can be very tedious at times with multi-line comments.

Reply:
I know @swiftal64 has had thoughts about this in the past too.

Reply:
Given that we say to wrap lines at 100 characters without any clarification on whether they are code lines or comments lines, we could also possibly avoid a new rule and simply add wrapping to `blockComments` in SwiftFormat.

Reply:
Yeah, it would be great to have a rule that enforces comments are <100 columns and automatically reflows comments onto new lines. Hopefully that would make it so that you could just write long block comments and then have the linter take care of reflowing them, and in Cal's case he could make them shorter as well.